src/main/java/bio/terra/service/resourcemanagement/azure/AzureAuthService.java

```diff
@@ -142,13 +153,31 @@ public BlobServiceClient getBlobServiceClient(
   /** Obtain a secret key for the associated storage account */
   private String getStorageAccountKey(
       UUID subscriptionId, String resourceGroupName, String storageAccountResourceName) {
-    AzureResourceManager client = configuration.getClient(subscriptionId);
 
-    return client
-        .storageAccounts()
-        .getByResourceGroup(resourceGroupName, storageAccountResourceName)
-        .getKeys()
-        .get(0)
-        .value();
+    int timeoutSeconds = 86400;
+    AzureAuthorizedCacheKey authorizedCacheKey =
+        new AzureAuthorizedCacheKey(subscriptionId, resourceGroupName, storageAccountResourceName);
+    AzureAuthorizedCacheValue authorizedCacheValue = authorizedMap.get(authorizedCacheKey);
+    if (authorizedCacheValue != null) { // check if it's in the cache
+      // check if it's still in the allotted time
+      if (Instant.now().isBefore(authorizedCacheValue.getTimeout())) {
```

Comment:
Ah, you can actually do this within an expiring cache (see https://github.com/DataBiosphere/jade-data-repo/blob/ad4d1b89b6825bac3c2d67aa1e436b5e0712366f/src/main/java/bio/terra/service/filedata/google/gcs/GcsPdao.java#L80)